iceoryx_integrationtest/iceoryx_integrationtest/test_user_header_example.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+import unittest
+
+import launch
+from launch_ros.substitutions import ExecutableInPackage
+import launch_testing
+import launch_testing.actions
+from launch_testing.asserts import assertSequentialStdout
+
+import pytest
+
+# @brief Test goal: "Integrationtest for the user-header example of iceoryx"
+# @pre setup ROS2 launch executables for RouDi (debug mode) and the example processes
+# @post check if all applications return exitcode 0 (success) after test run
+@pytest.mark.launch_test
+def generate_test_description():
+
+    proc_env = os.environ.copy()
+    colcon_prefix_path = os.environ.get('COLCON_PREFIX_PATH', '')
+
+    roudi_executable = os.path.join(
+        colcon_prefix_path,
+        'iceoryx_posh/bin/',
+        'iox-roudi'
+    )
+    roudi_process = launch.actions.ExecuteProcess(
+        cmd=[roudi_executable, '-l', 'debug'],
+        env=proc_env, output='screen',
+        sigterm_timeout='20')
+
+    user_header_subscriber_executable = os.path.join(
+        colcon_prefix_path,
+        'example_user_header/bin/',
+        'iox-cpp-user-header-subscriber'
+    )
+    user_header_subscriber_process = launch.actions.ExecuteProcess(
+        cmd=[user_header_subscriber_executable],
+        env=proc_env, output='screen')
+
+    user_header_publisher_executable = os.path.join(
+        colcon_prefix_path,
+        'example_user_header/bin/',
+        'iox-cpp-user-header-publisher'
```

Comment:
@dkroenke the example has executables for the typed/untyped C++ and C API, which means I would need to create three different files. Can I do everything in one file?

Reply:
@elBoberido Yes that's possible. Due to the fact that all applications have different outputs you can start all processes in parallel and check on them.
 Take a look at https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_integrationtest/iceoryx_integrationtest/test_icedelivery_example.py to get an idea how to handle a bunch of processes.

Reply:
that would be possible with the subscriber but the publisher would publish on the same topic. Don't know it that's a good idea

Reply:
You can separate the topics for c and cxx and the problem is resolved. Or you need to create a separate file....

Reply:
If the topics are separated, it's not possible anymore to mix the implementations. Is there no way to have multiple test within one file?

Reply:
okay, I just initialized the timestamp in each implementation with a different value. This makes it possible to distinguish the implementations and mix them for the tests